File: Codes/Bingran Li/Tree/226. Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode invertTree(TreeNode root) {
        
       if(root==null|| (root.left==null && root.right==null)) // add a judgement to reduce operation, reduce from 1ms to 0ms
            return root;
            
        TreeNode temNode = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(temNode);
  
        return root;
    }
 
}

File: Codes/Yangkun Huang/Tree/226. Invert Binary Tree.cpp
TreeNode* invertTree(TreeNode* root) {
        if(root)
        {
            invertTree(root->left);  // postorder traversal
            invertTree(root->right);
            TreeNode* temp=root->left;
            root->left=root->right;
            root->right=temp;
        }
        return root;
    }